src/planner/auto-layout.ts

```diff
@@ -0,0 +1,247 @@
+import { BlockDefinition, Plan } from '@kapeta/schemas';
+import { AssetInfo, PlannerNodeSize } from '../types';
+import { parseKapetaUri } from '@kapeta/nodejs-utils';
+import { calculateBlockSize } from './BlockContext';
+import { BlockMode } from '../utils/enums';
+import { BlockTypeProvider } from '@kapeta/ui-web-context';
+import { BLOCK_SIZE } from './utils/planUtils';
+
+const DEFAULT_LAYOUT_MARGIN_X = 500;
+const DEFAULT_LAYOUT_MARGIN_Y = 100;
+const NODE_WIDTH = BLOCK_SIZE;
+const NODE_HEIGHT = BLOCK_SIZE;
+const CANVAS_OFFSET_LEFT = 250;
+const CANVAS_OFFSET_TOP = 0;
+
+interface LayoutOptions {
+    nodeSize: PlannerNodeSize;
+    nodeWidth: number;
+    marginX: number;
+    marginY: number;
+}
+
+function toEdgeId(providerId: string, consumerId: string) {
+    return [providerId, consumerId].sort().join('-');
+}
+
+export function applyAutoLayout(
```

Comment:
Can you add a description of the general idea of the layout logic? Will probably make reading the code a lot easier for me.

Reply:
Done

Reply:
On it ✏️